RevenueCatUI/CustomerCenter/Abstractions/CustomerCenterPurchasesType.swift

```diff
@@ -11,6 +11,8 @@
 //
 //  Created by Cesar de la Vega on 18/7/24.
 
+#if CUSTOMER_CENTER_ENABLED
+
 import Foundation
```

Comment:
okay, I'll admit that it's annoying to have this in 30 files, but it legitimately took me just a couple of mins to add

Reply:
and we'll only need to remove it when we're ready to ship the feature

Reply:
I bet we can probably leave it out in some of these files... But better to be sure 👍

Reply:
No idea if this is in the realm of possibilities, but would it be possible to have the the flag only around public entry points of the Customer Center feature? E.g. `Purchases.loadCustomerCenter()`, `CustomerCenterView()` and the `presentCustomerCenter()` modifier?

Reply:
While that should be fine for `RevenueCat` (we would just need the flag for `loadCustomerCenter` and `CustomerCenterConfigData`), since `CustomerCenterConfigData` is used throughout `RevenueCatUI`, we would need to wrap many files there.

Reply:
What if.. we only wrap the constructor of `CustomerCenterConfigData` (or make the constructor `public` if the flag exists, `internal` otherwise). We'd be exposing some extra types, but no dev would be able to actually do anything with them because they can't construct anything. 
(Just spitballing, lmk if this is dumb.)

Reply:
There are a bunch of internal types inside `CustomerCenterConfigData` so we would need to wrap a bunch of constructors... It would however, remove the need to wrap the code in `RevenueCatUI`... 

Having said that, we would still be exposing new types which can potentially change, even if they are unusable... So I don't think it's ideal 😓

Reply:
Hmm yea, not ideal indeed. 🤔

Reply:
Yeah I think you're making very good points. Not worth the extra effort to do fine tuning imo. Also, we don't have a good idea what fine tuning would even look like yet.